### ISSUE TITLE
router-view in dialog not working every time

### ISSUE THREAD

**Issue by vinayhfs** [aurelia/framework#808]
* **Operating System:**
Windows 7

* **Node Version:**
6.10.3

* **NPM Version:**
3.10.10

Aurelia CLI Version*
0.27.0

* **Browser:**
Chrome 59.0

* **Language:**
TypeScript 

**Current behavior:**
if we use router-view in dialog on first opening of dialog routing works well and can see view but if we close dialog and reopen dialog, view content is not visible. 
Observation is view model instance is not getting created second time

**Expected/desired behavior:**
Every time we open dialog routing should work

Steps to reproduce - Code details- 
I just modified Contact Manager Tutorial. 
Added aurelia-dialog to project and configured.
Routing app.ts is kept as it is. Added dialog open code in app.ts and in app.html added new button. 
On this button click opened dialog.  From app.html removed 2 lines of contact list and router-view tags and added in dialog prompt body part.
Run the code which will open dialog box - will show contact details or "please select contact" depends on state of app.
Now close dialog and click button re open dialog. it will show you list of contacts on left side part but details on right side are missing.

Attached sample code. just do npm install and run application.

[contact-manager.zip](https://github.com/aurelia/framework/files/1175636/contact-manager.zip)

**Comment by Alexander-Taran** [comment]
@EisenbergEffect looks like not an envisioned use case. Yet almighty framework lets you do stuff like this.
Need an opinion on supporting router-view inside dialog

can't imagine a deep linking scenario for this one though

**Comment by EisenbergEffect** [comment]
This is not an intended use case. @davismj to comment on whether or not he wants to take this on. It seems very tricky to determine how to make this work all the time.

**Comment by davismj** [comment]
I'm going to guess this is a dialog issue and the dialog would need to be modified to support this.

@vinayhfs In your case, please consider building a dialog component rather than relying on a library. It might be easier than you think. See my article here: http://davismj.me/blog/easy-dialog/

**Comment by EisenbergEffect** [comment]
I don't think we should make special support for this. It's a very particular case and not that common. I'm not sure if it's a good idea or not. Also, it's still possible to implement on top of the router by end users, I would think. You would just use a wildcard route for your dialog routes and then handle the fragment manually in your wizard. Or you could use special query strings that were interpreted directly by the wizard.

**Comment by Alexander-Taran** [comment]
close as won't-fix?

**Comment by davismj** [comment]
@EisenbergEffect Yes, please close as wont-fix.
